src/pubsub.c

```diff
@@ -105,19 +105,22 @@ pubsubtype pubSubShardType = {
  * to send a special message (for instance an Array type) by using the
  * addReply*() API family. */
 void addReplyPubsubMessage(client *c, robj *channel, robj *msg, robj *message_bulk) {
+    c->flags |= CLIENT_PUSHING;
     if (c->resp == 2)
         addReply(c,shared.mbulkhdr[3]);
     else
         addReplyPushLen(c,3);
     addReply(c,message_bulk);
     addReplyBulk(c,channel);
     if (msg) addReplyBulk(c,msg);
+    c->flags &= ~CLIENT_PUSHING;
```

Comment:
maybe we better backup and restore the flags?

Reply:
you mean do a `old_flags` and then `|=` it in everywhere? it look like a lot change will make

Reply:
i mean that if the PUSHING flag was already set when we entered this function, we should not unset it.
so either backup and restore the flags, or just remember the old value of this flag and avoid removing it at the end.

Reply:
i suppose that restoring the full original flags is wrong, since the executed functions can set flags (like CLOSE_ASAP), so we just need to remember if the PUSHING flag was set, and then avoid unseting it.